app/repositories/repositories.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import play.api.libs.json.JsResultException
+
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package object repositories {
+  def hasDupeKeyViolation(ex: JsResultException): Boolean = (for {
+    validationErrors <- ex.errors.flatMap(_._2)
+    message <- validationErrors.messages
+    dupeKey = message.matches(".*code=11000[^\\w\\d].*")
```

Comment:
What is the purpose of the part of the regex [^\\w\\d]; I think ".*code=11000.*" will also work. Either way this test is dependent on the error message raised by Mongo which may change between versions of the database. As such it may be better to store the regex expression in the application configuration file.

Reply:
I got this from doing a search online.  Changing an error code would be a major change that will impact almost every mongo based app developed, so I really doubt that will ever change.  Worst case, it woudl return a bad request/server error if that code ever changed

Reply:
Yes, I see that the general error is logged so that we would be able to work out what was going wrong.